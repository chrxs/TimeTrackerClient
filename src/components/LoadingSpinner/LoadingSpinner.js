import React from 'react'
import PropTypes from 'prop-types'

import styles from './LoadingSpinner.scss'

const LoadingSpinner = ({
  size,
  strokeWidth,
  centered,
  style,
  ...rest
}) => {
  let loadingSpinnerStyle = {
    width: `${size}px`
  }

  if (centered) {
    loadingSpinnerStyle.position = 'absolute'
    loadingSpinnerStyle.top = '50%'
    loadingSpinnerStyle.left = '50%'
    loadingSpinnerStyle.margin = `-${size / 2}px 0 0 -${size / 2}px`
  }

  loadingSpinnerStyle = {
    ...loadingSpinnerStyle,
    ...style
  }

  return (
    <div
      className={styles.LoadingSpinner}
      style={loadingSpinnerStyle}
      {...rest}
    >
      <svg
        className={styles.circular}
        viewBox='25 25 50 50'
      >
        <circle
          className={styles.path}
          cx='50'
          cy='50'
          r='20'
          fill='none'
          strokeWidth={strokeWidth}
          strokeMiterlimit='10'
        />
      </svg>
    </div>
  )
}

LoadingSpinner.displayName = 'LoadingSpinner'

LoadingSpinner.propTypes = {
  size: PropTypes.number,
  strokeWidth: PropTypes.number,
  centered: PropTypes.bool,
  style: PropTypes.object
}

LoadingSpinner.defaultProps = {
  size: 100,
  strokeWidth: 2,
  centered: false,
  style: {}
}

export default LoadingSpinner
