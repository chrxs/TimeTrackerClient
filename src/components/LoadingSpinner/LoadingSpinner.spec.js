/* global describe, test, expect */
import React from 'react'
import LoadingSpinner from './LoadingSpinner'
import renderer from 'react-test-renderer'

describe('LoadingSpinner snapshots', () => {
  test('with just label prop', () => {
    const component = renderer.create(<LoadingSpinner />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with all props', () => {
    const component = renderer.create(<LoadingSpinner size={200} strokeWidth={4} centered />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
