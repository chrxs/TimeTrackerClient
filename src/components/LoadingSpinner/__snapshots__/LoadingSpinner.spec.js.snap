// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner snapshots with all props 1`] = `
<div
  className="LoadingSpinner"
  style={
    Object {
      "left": "50%",
      "margin": "-100px 0 0 -100px",
      "position": "absolute",
      "top": "50%",
      "width": "200px",
    }
  }
>
  <svg
    className="circular"
    viewBox="25 25 50 50"
  >
    <circle
      className="path"
      cx="50"
      cy="50"
      fill="none"
      r="20"
      strokeMiterlimit="10"
      strokeWidth={4}
    />
  </svg>
</div>
`;

exports[`LoadingSpinner snapshots with just label prop 1`] = `
<div
  className="LoadingSpinner"
  style={
    Object {
      "width": "100px",
    }
  }
>
  <svg
    className="circular"
    viewBox="25 25 50 50"
  >
    <circle
      className="path"
      cx="50"
      cy="50"
      fill="none"
      r="20"
      strokeMiterlimit="10"
      strokeWidth={2}
    />
  </svg>
</div>
`;
